docs/src/modules/java/pages/agents.adoc

```diff
@@ -341,23 +341,42 @@ You may frequently hear people say things like "the LLM can make a call" or "the
 
 The Agent will then in turn execute the tool requested by the LLM, incorporate the tool results into the session context, and then send a new prompt. This will continue in a loop until the LLM no longer indicates it needs to invoke a tool to perform its task.
 
-Tool functions are incredibly easy in Akka. All you need to do is decorate an Agent method with the `@FunctionTool`
-annotation, and Akka will automatically resolve it on your behalf. As an example, you might have an agent that
-recommends outdoor activities, but to do so it needs to know the current weather.
+There are two ways to add function tools to your agent:
 
-Anything "current" is beyond the capabilities of an LLM, so the model response will include information on what tool needs to be invoked.
+1. **Agent-defined function tools** — Define function tools directly within your agent class using the`@FunctionTool`
+annotation. These are automatically registered as available tools.
 
-Here's a sample `getWeather` tool function. The method is defined in the `WeatherAgent` class. Since the method is
-defined in the agent itself, Akka will automatically register it as a tool function available to this agent.
+2. **Externally defined function tools** — Explicitly register external objects or classes containing function tools by
+passing them to the `effects().tools()` method in your agent's command handler. Objects or classes passed to `effects
+().tools()` must have at least one public method annotated with `@FunctionTool`.
+
+You can use either approach independently or combine them based on your needs. Let's look at a complete example showing both approaches:
```

Comment:
very cool that we can combine them. what happens if this produces a duplicate tool name, one defined locally and one defined in an external class? Does this error or does one take precedence/override?

Reply:
There is a fixme about it. We need to use full qualified names (full class name + method name) to avoid clashes. 

Good that you mentioned, because we need to have it in place as soon as possible